ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/TableExport.java

```diff
@@ -0,0 +1,271 @@
+          if (replicationSpec.isMetadataOnly()) {
+            return null;
+          } else {
+            return new PartitionIterable(db, tableSpec.tableHandle, null, conf.getIntVar(
+                HiveConf.ConfVars.METASTORE_BATCH_RETRIEVE_MAX));
+          }
+        } else {
+          // PARTITIONS specified - partitions inside tableSpec
+          return new PartitionIterable(tableSpec.partitions);
+        }
+      } else {
+        // Either tableHandle isn't partitioned => null, or repl-export after ts becomes null => null.
+        // or this is a noop-replication export, so we can skip looking at ptns.
+        return null;
+      }
+    } catch (Exception e) {
+      throw new SemanticException("Error when identifying partitions", e);
+    }
+  }
+
+  private void writeMetaData(PartitionIterable partitions) throws SemanticException {
+    try {
+      EximUtil.createExportDump(
+          paths.exportFileSystem,
+          paths.metaDataExportFile(),
+          tableSpec == null ? null : tableSpec.tableHandle,
+          partitions,
+          replicationSpec);
+      logger.debug("_metadata file written into " + paths.metaDataExportFile().toString());
+    } catch (Exception e) {
+      // the path used above should not be used on a second try as each dump request is written to a unique location.
+      // however if we want to keep the dump location clean we might want to delete the paths
+      throw new SemanticException(
+          ErrorMsg.IO_ERROR.getMsg("Exception while writing out the local file"), e);
+    }
+  }
+
+  private void writeData(PartitionIterable partitions) throws SemanticException {
+    try {
+      if (tableSpec.tableHandle.isPartitioned()) {
+        if (partitions == null) {
+          throw new IllegalStateException(
+              "partitions cannot be null for partitionTable :" + tableSpec.tableName);
+        }
+        for (Partition partition : partitions) {
+          Path fromPath = partition.getDataLocation();
+          // this the data copy
+          Path rootDataDumpDir = paths.partitionExportDir(partition.getName());
+          new FileOperations(fromPath, rootDataDumpDir, conf).export(replicationSpec);
+          authEntities.inputs.add(new ReadEntity(partition));
+        }
+      } else {
+        Path fromPath = tableSpec.tableHandle.getDataLocation();
+        //this is the data copy
+        new FileOperations(fromPath, paths.dataExportDir(), conf).export(replicationSpec);
+        authEntities.inputs.add(new ReadEntity(tableSpec.tableHandle));
+      }
+      authEntities.outputs.add(toWriteEntity(paths.exportRootDir, conf));
+    } catch (Exception e) {
+      throw new SemanticException(e);
+    }
+  }
+
+  private boolean shouldExport() throws SemanticException {
+    if (replicationSpec.isInReplicationScope()) {
+      return !(tableSpec.tableHandle.isTemporary() || tableSpec.tableHandle.isNonNative());
```

Comment:
No we cant avoid this as the fact that tablespec is null handled separately with writing only metadata for the condition that is in constructor and handling of this condition is done separately post that.